dev-libs/qtcompress/qtcompress-0_p20180513.ebuild

```diff
@@ -0,0 +1,59 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit qmake-utils toolchain-funcs virtualx
+
+DESCRIPTION="A public API for QZip in an easy to use module"
+
+HOMEPAGE="https://github.com/nezticle/qtcompress"
+MY_COMMIT="23f8831826cd72aedf99fc3699148b6c994fd677"
+SRC_URI="https://github.com/nezticle/qtcompress/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+LICENSE="LGPL-2.1"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+S="${WORKDIR}/qtcompress-${MY_COMMIT}"
+
+RDEPEND="
+	dev-qt/qtcore:5
+	sys-libs/zlib:=
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	test? ( dev-qt/qttest:5 )
+"
+
+PATCHES=(
+	"${FILESDIR}/${P}-remove-zlib.patch"
+	"${FILESDIR}/${P}-test-include.patch"
+)
+
+src_prepare() {
+	default
+	rm -r src/3rdparty/ || die
+}
+
+src_configure() {
+	local pkg_config="$(tc-getPKG_CONFIG)"
+	eqmake5 \
+		"INCLUDEPATH+=$("${pkg_config}" --cflags zlib)" \
+		"LIBS+=$("${pkg_config}" --libs zlib)"
+}
+
+src_test() {
+	cd tests/auto/unit || die
+	eqmake5 \
+		"INCLUDEPATH+=${S}/src/compress"
+	emake
+	virtx qzip/target_wrapper.sh qzip/tst_qzip || die
```

Comment:
`virtx` is a `nonfatal` function so don't add `|| die` here at all.
https://devmanual.gentoo.org/ebuild-writing/error-handling/index.html
https://devmanual.gentoo.org/eclass-reference/virtualx.eclass/index.html

Basically it opens a new Xvfb session and if you kill the command, the X session may still be left open.

Reply:
||die applies to the whole virtx command, not the part of it wrapped by virtx, so ||die here is harmless. The guide means that the wrapped command should not die. But sure.